### PR TITLE
Signal reservation updates #852614871284588

### PR DESCRIPTION
**Pull request by jessehon**
https://app.asana.com/0/683768070860536/852614871284588

### CODE REVIEW

File: app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    axios
+      .patch(SIGNALS_RESERVATION_URL, this.state.formData, { headers })
+      .then(() => {
+        this.setState((state) => ({
+          activeStep: state.activeStep + 1,
+        }))
+      })
+      .catch((error) => {
+        // TODO: Handle error by showing some sort of message.
+        alert(error)
```
Comment (jessehon) [inline app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx]: Are we happy to leave this as an `alert`?

File: app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    return haveValidationsPassed
+  }
+
+  public handleNext = () => {
+    if (!this.validateStep(this.state.activeStep)) {
+      return
+    }
+
+    const headers = {
+      'X-CSRF-Token': document
```
Comment (jessehon) [inline app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx]: It is better to pass this in via a prop than try to assume there would always be this element. There is an example of how I do this in the `NavUser` component and the logout button

File: app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx (hunk 1 above)
Comment (hanchang) [reply app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx]: Yeah, I was going to use Material UI popup but figured it's overkill for now, but we should go back and tidy it up later.

Comment (jessehon) [reply app/javascript/App/bundles/signalsPage/SignalReservationForm.tsx]: Sure

**Pull request merged**